libvirt/resource_libvirt_domain.go

```diff
@@ -267,6 +289,25 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if kernel, ok := d.GetOk("kernel"); ok {
```

Comment:
we could wrap this new code in a function. because the DomainCreate is a really big function.